lib/iris/coords.py

```diff
@@ -464,24 +464,40 @@ def __getitem__(self, key):
         # each dimension of the coord.
         full_slice = iris.util._build_full_slice_given_keys(key, self.ndim)
 
-        # If it's a "null" indexing operation (e.g. coord[:, :]) then
-        # we can preserve deferred loading by avoiding promoting _points
-        # and _bounds to full ndarray instances.
+        # Fetch the points and bounds.
+        points = self._points
+        bounds = self._bounds
+
+        # Track whether we need to copy data after any indexing operation.
+        # NOTE: when we get rid of LazyArray-s, we can remove this bit.
+        points_copied = False
+        bounds_copied = False
+
+        # If it's a "null" indexing operation (e.g. coord[:, :]) then we can
+        # avoid all the indexing.  This is desirable if we have LazyArray-s,
+        # as they must be fetched first (you can't index them).
         def is_full_slice(s):
             return isinstance(s, slice) and s == slice(None, None)
-        if all(is_full_slice(s) for s in full_slice):
-            points = self._points
-            bounds = self._bounds
+        null_operation = all(is_full_slice(s) for s in full_slice)
+
+        if null_operation:
+            # If we have Lazy Arrays, don't copy but just allow the new coord
+            # to share the same LazyArray object.
+            if isinstance(points, iris.aux_factory._LazyArray):
+                points_copied = True
+            if bounds is not None:
+                if isinstance(bounds, iris.aux_factory._LazyArray):
+                    bounds_copied = True
```

Comment:
I don't know how fussy you want to be about naming, but to me this `bounds_copied` says that the action has been done. `copy_bounds = True` makes more sense to me